### PR TITLE
Adding hint/warning log for use under a MultiChildrenderObjectElement

### PR DESCRIPTION
**Pull request by pstromberg98**
Adds error when Nil is used "improperly"

### CODE REVIEW

**Comment by pstromberg98** [comment]
@letsar One thing I do worry about using an assert is that it throws. I think it would be safer if we just made it warn instead. Because the code is essentially assuming that all `MultiChildRenderObjectElement` corresponding widgets take lists as arguments (which they might, I don't know), and thus can use inline conditionals or list modifications. Is that assumption safe to make?

**Comment by letsar** [comment]
Thank you a lot for this!

You can make an assert than does not throw like this:
```
assert(() {
  // You can log here what you want.
  return true;
}());
```

I'm not sure a warn is safer, because the developer will maybe not read it. For the moment , until https://github.com/flutter/flutter/issues/78159 is fixed, I think we should prevent the developer to use it in this context.

What do you think?

**Comment by letsar** [comment]
LGTM by the way :-)

**Review by letsar — COMMENTED** [review]
Great! Can you also increment the version to 1.1.0 and add an entry in the changelog?

**Comment by pstromberg98** [comment]
Yeah my only worry is if there is a situation in which a MultiChildRenderObjectElement is created from a widget that doesn't take in a list. But I don't think that there are any widget-element pairs like that so I am fine having it throw if you are.

**Comment by letsar** [comment]
In fact it would be even more important in that context, don't you think?

**Comment by pstromberg98** [comment]
@letsar Yes I suppose so. Because they could just use Nil instead of the hypothetical widget with a Nil child?

Regardless, looking at the documentation the `MultiChildRenderObjectElement` requires a `MultiChildRenderObjectWidget` (which takes a list) so I don't think my hypothetical example would ever actually happen.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.1.1]
```
Comment (letsar) [inline CHANGELOG.md:10]: Can you also add this link below [Unreleased]?

`[1.1.1]: https://github.com/letsar/binder/compare/releases/tag/v1.1.1`

**Review by letsar — CHANGES_REQUESTED** [review]
Great, thanks again :-). Just a minor change in the changelog, and I will accept this PR :-)

**Comment by pstromberg98** [comment]
@letsar Is this going to be merged still?

**Review by letsar — APPROVED** [review]
LGTM

**Comment by letsar** [comment]
Merged and published. Thank you again Parker :-)

**Pull request merged**